tests/rptest/archival/s3_client.py

```diff
@@ -458,3 +471,7 @@ def list_buckets(self, client=None) -> dict[str, Union[list, dict]]:
         except Exception as ex:
             self.logger.error(f'Error listing buckets: {ex}')
             raise
+
+    def _add_header(self, model, params, request_signer, **kwargs):
```

Comment:
nit, since only params is used, the method signature can be updated:
```suggestion
    def _add_header(self, params):
```

Reply:
I see this is used as a handler, so perhaps the extra params are required for the handler api?

Reply:
right those are required when before-call is used